lib/Serialization/Serialization.cpp

```diff
@@ -3677,14 +3686,23 @@ static TypeAliasDecl *findTypeAliasForBuiltin(ASTContext &Ctx, Type T) {
 
 void Serializer::writeType(Type ty) {
   using namespace decls_block;
+  PrettyStackTraceType traceRAII(ty->getASTContext(), "serializing", ty);
 
   auto id = DeclAndTypeIDs[ty];
   assert(id != 0 && "type not referenced properly");
   (void)id;
 
   assert((id - 1) == TypeOffsets.size());
-
+  assert((DeclOffsets.empty() || DeclOffsets.back() != Out.GetCurrentBitNo()) &&
+         "encoding Decl and Type to the same offset");
   TypeOffsets.push_back(Out.GetCurrentBitNo());
+  SWIFT_DEFER {
```

Comment:
If this is the last statement in the function, why `SWIFT_DEFER` it?

Reply:
It's…not?

Reply:
🤦‍♂️ misread the braces, apologies.